### PR TITLE
Add pretrained markov abstractions to dev

### PR DESCRIPTION
**Pull request by camall3n**

### CODE REVIEW

File: env.py [hunk 1]
```diff
@@ -33,22 +32,13 @@ def make_ari_onehot(env, num_frames, training=False):
     return FrameStack(ResetARIOneHot(AtariARIWrapper(env)), num_frames)
 
 
-def get_env(args):
+def get_train_test_envs(args):
```
Comment (camall3n) [inline env.py:35]: Renamed for clarity

File: env.py [hunk 2]
```diff
@@ -62,12 +52,21 @@ def get_env(args):
...
+
+def get_run_tag(args):
+    # Get uuid for run
+    if args.uuid == 'env':
+        uuid_tag = args.env
+    elif args.uuid == 'random':
+        import uuid
+        uuid_tag = uuid.uuid4()
+    else:
+        uuid_tag = args.uuid
```
Comment (camall3n) [inline env.py:65]: Moved uuid stuff to its own method

File: main.py [hunk 3]
```diff
@@ -128,24 +127,6 @@ def log(s):
...
-
-def save_memory(memory, memory_path, disable_bzip):
-    if disable_bzip:
-        with open(memory_path, 'wb') as pickle_file:
-            pickle.dump(memory, pickle_file)
-    else:
-        with bz2.open(memory_path, 'wb') as zipped_pickle_file:
-            pickle.dump(memory, zipped_pickle_file)
-
-
```
Comment (camall3n) [inline main.py:148]: Moved this load_memory and save_memory to memory.py

File: memory.py [hunk 4]
```diff
@@ -1,12 +1,30 @@
...
+            return pickle.load(zipped_pickle_file)
+
+def save_memory(memory, memory_path, disable_bzip):
+    if disable_bzip:
+        with open(memory_path, 'wb') as pickle_file:
+            pickle.dump(memory, pickle_file)
+    else:
+        with bz2.open(memory_path, 'wb') as zipped_pickle_file:
+            pickle.dump(memory, zipped_pickle_file)
+
```
Comment (camall3n) [inline memory.py:27]: Moved load_memory and save_memory from main.py

File: memory.py [hunk 5]
```diff
@@ -118,12 +136,10 @@ def _get_sample_from_segment(self, segment, i):
...
-            sample = np.random.uniform(
-                i * segment, (i + 1) * segment
-            )  # Uniformly sample an element from within a segment
-            prob, idx, tree_idx = self.transitions.find(
-                sample
-            )  # Retrieve sample from tree with un-normalised probability
+            # Uniformly sample an element from within a segment
+            sample = np.random.uniform(i * segment, (i + 1) * segment)
+            # Retrieve sample from tree with un-normalised probability
+            prob, idx, tree_idx = self.transitions.find(sample)
```
Comment (camall3n) [inline memory.py:142]: Cleanup

File: memory.py [hunk 6]
```diff
@@ -163,22 +179,23 @@ def _get_sample_from_segment(self, segment, i):
...
+        probs = np.array(probs, dtype=np.float32) / p_total
         capacity = self.capacity if self.transitions.full else self.transitions.index
-        weights = (capacity * probs)**-self.priority_weight  # Compute importance-sampling weights w
-        weights = torch.tensor(
-            weights / weights.max(), dtype=torch.float32, device=self.device
-        )  # Normalise by max importance-sampling weight from batch
+        # Compute importance-sampling weights w
+        weights = (capacity * probs)**-self.priority_weight
+        # Normalise by max importance-sampling weight from batch
+        weights = torch.tensor( weights / weights.max(), dtype=torch.float32, device=self.device)
```
Comment (camall3n) [inline memory.py:198]: More cleanup

File: model.py [hunk 7]
```diff
@@ -70,14 +70,57 @@ def forward(self, input):
         else:
             return F.linear(input, self.weight_mu, self.bias_mu)
 
+def build_phi_network(args, input_shape=None):
```
Comment (camall3n) [inline model.py:73]: pulled this out of DQN class

**Review by neevparikh — APPROVED** [review]

**Pull request merged**